tools/package_executable.py

```diff
@@ -0,0 +1,129 @@
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    # create symbolic links
+    if sys.platform == "win32":
+        cmd = ["conda", "activate", f"{args.conda_env}&", "python", "create_symbolic_links.py"]
+    elif sys.platform in ["darwin", "linux"]:
+        cmd = [f". {args.conda_path}; conda activate {args.conda_env}; python create_symbolic_links.py"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    # generate config file
+    if sys.platform == "win32":
+       cmd = ["conda", "activate", f"{args.conda_env}&", "python", "generate_config.py"]
+    elif sys.platform in ["darwin", "linux"]:
+        cmd = [f". {args.conda_path}; conda activate {args.conda_env}; python generate_config.py"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    # build the executable
+    if sys.platform == "win32":
+        build_exe = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.bat")
+    elif sys.platform in ["darwin", "linux"]:
+        build_exe = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+    else:
+        assert False
+    
+    cmd = [
+        build_exe,
+        "BuildCookRun",
+        "-project=" + os.path.join(repo_dir, "cpp", "unreal_projects", "SpearSim", "SpearSim.uproject"), 
+        "-build",
+        "-cook",
+        "-stage",
+        "-package",
+        "-archive",
+        "-pak",
+        "-targetplatform=" + args.target_platform,
+        "-target=SpearSim",
+        "-archivedirectory=" + os.path.join(args.temp_dir, f"SpearSim-{args.config_mode}"),
+        "-clientconfig=" + args.config_mode
+    ]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    # change working directory
+    print(f"[SPEAR | build.py] Changing working directory to {cwd}")
+    os.chdir(cwd)
+
+    print(f"[SPEAR | build.py] Successfully built SpearSim for {args.target_platform} platform with {args.config_mode} mode.")
+
+    # zip the exectuable for distribution
+    if sys.platform == "win32":
+        platform_dir_name = "WindowsNoEditor"
+    elif sys.platform == "darwin":
+        platform_dir_name = "MacNoEditor"
+    elif sys.platform == "linux":
+        platform_dir_name = "LinuxNoEditor"
+    else:
+        assert False
+    
+    os.makedirs(args.output_dir, exist_ok=True)
+
+    # TODO: need to codesign on macOS
+    file = shutil.make_archive(base_name=os.path.join(args.output_dir, f"SpearSim-{args.output_tag}-{args.target_platform}-{args.config_mode}"), format='zip', root_dir=os.path.join(args.temp_dir, f"SpearSim-{args.config_mode}", platform_dir_name))
```

Comment:
- prefer double-quotes, e.g., format="zip"
- form the whole path in the sys.platform if statement, like you do for run_uat_script so that you don't need to call os.path.join inline when calling make_archive

Reply:
Since we are removing packaging in this script, we no longer need this step